Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLParameterScan.py

```diff
@@ -0,0 +1,242 @@
+                                               optional=PropertyMode.Optional),
+                             doc="The output workspace containing all the reduced data, before grouping.")
+
+        self.declareProperty(MultipleFileProperty('SampleRuns',
+                                                  action=FileAction.Load,
+                                                  extensions=['nxs']),
+                             doc='Sample run(s).')
+
+        self.declareProperty(MultipleFileProperty('AbsorberRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Absorber (Cd/B4C) run(s).')
+
+        self.declareProperty(MultipleFileProperty('ContainerRuns',
+                                                  action=FileAction.OptionalLoad,
+                                                  extensions=['nxs']),
+                             doc='Empty container run(s).')
+
+        self.setPropertyGroup('SampleRuns', 'Numors')
+        self.setPropertyGroup('AbsorberRuns', 'Numors')
+        self.setPropertyGroup('ContainerRuns', 'Numors')
+
+        self.declareProperty(FileProperty('SensitivityMap', '', action=FileAction.OptionalLoad, extensions=['nxs']),
+                             doc='File containing the map of relative detector efficiencies.')
+
+        self.declareProperty(FileProperty('DefaultMaskFile', '', action=FileAction.OptionalLoad, extensions=['nxs']),
+                             doc='File containing the default mask to be applied to all the detector configurations.')
+
+        self.copyProperties('SANSILLReduction', ['NormaliseBy'])
+
+        self.declareProperty('Observable', 'Omega.value',
+                             doc='Parameter from the sample logs along which the scan is made')
+
+        self.declareProperty('PixelYMin', 140, validator=IntBoundedValidator(lower=0),
+                             doc='Minimal y-index taken in the integration')
+        self.declareProperty('PixelYMax', 180, validator=IntBoundedValidator(lower=0),
+                             doc='Maximal y-index taken in the integration')
+
+        self.declareProperty('Wavelength', 0., validator=FloatBoundedValidator(lower=0.),
+                             doc='Wavelength of the experiment. Will try to read Nexus files if not provided.')
+
+        self.setPropertyGroup('SensitivityMap', 'Options')
+        self.setPropertyGroup('DefaultMaskFile', 'Options')
+        self.setPropertyGroup('NormaliseBy', 'Options')
+        self.setPropertyGroup('Observable', 'Options')
+        self.setPropertyGroup('PixelYMin', 'Options')
+        self.setPropertyGroup('PixelYMax', 'Options')
+
+    def PyExec(self):
+
+        self.setUp()
+
+        _, load_ws_name = needs_loading(self.sample, "Load")
+        self.progress.report(7, 'Loading samples')
```

Comment:
I'm not sure this line is correct, what does 7 come from? This causes the progress bar jump at 7 and then back to zero again.